src/Microsoft.Health.Fhir.SqlServer/Features/Storage/Registry/SqlServerSearchParameterStatusDataStore.cs

```diff
@@ -50,7 +50,7 @@ public async Task<IReadOnlyCollection<ResourceSearchParameterStatus>> GetSearchP
                 {
                     while (await sqlDataReader.ReadAsync())
                     {
-                        (string uri, string stringStatus, DateTimeOffset lastUpdated, bool isPartiallySupported) = sqlDataReader.ReadRow(
+                        (string uri, string stringStatus, DateTimeOffset? lastUpdated, bool? isPartiallySupported) = sqlDataReader.ReadRow(
```

Comment:
Just wondering, if status in the SearchParam is null and when we try to Enum.Parse the null value it might results in 500 if not handled properly. Should we handle it as OperationOutcome?

Reply:
Good question! I discussed this with @feordin a bit, and I now throw an error in 9a9827c.

These columns are nullable because they are added to `dbo.SearchParam` in V4, after there is already existing data in the table.

They should be populated as soon as they are added to the table (in `SqlServerFhirModel.Start()`). If this check fails, we should be returning a 500 because something would be broken on our end that the user could not fix.

Reply:
Thanks for taking a look. It makes sense. As discussed, it is not required to throw NullReferenceException as a reduandant code in the mentioned commit.